### PR TITLE
*: Support tls for external #492

### LINKED ISSUE #492: [Feature] Transport Layer Security (TLS) cryptographic protocol
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zhl003**
### Which issue(s) this PR fixes?

Fixes #492

### What this PR does?

Summary:
Support tls for external

### CODE REVIEW

File: sidecar/init.go [hunk 1]
```diff
@@ -315,3 +317,18 @@ func buildDefaultXenonMeta(uid, gid int) error {
 	}
 	return nil
 }
+func buildSSLdata(uid, gid int) error {
```
Comment (acekingke) [inline sidecar/init.go:320]: uid , gid not used

File: sidecar/init.go [hunk 2]
```diff
@@ -315,3 +317,18 @@ func buildDefaultXenonMeta(uid, gid int) error {
...
+	// cp -rp /tmp/myssl/* /etc/mysql/ssl/
+	//refer https://stackoverflow.com/questions/31467153/golang-failed-exec-command-that-works-in-terminal
+	shellCmd := "cp  /tmp/mysql-ssl/* " + utils.TlsMountPath
+	cmd := exec.Command("sh", "-c", shellCmd)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to copy ssl: %s", err)
+	}
+	cronCmd := "chown -R mysql.mysql " + utils.TlsMountPath
```
Comment (acekingke) [inline sidecar/init.go:328]: add a blank line here

File: utils/constants.go [hunk 3]
```diff
@@ -47,6 +47,10 @@ var (
 		ClusterAdd:      http.MethodPost,
 		ClusterRemove:   http.MethodPost,
 	}
+	//RunAsUser is the user id for mysql
+	RunAsUser int64 = 1001
```
Comment (acekingke) [inline utils/constants.go:51]: not used, remove it,  if you want use, please call `user.Lookup("mysql")` is more safe.

File: mysqlcluster/mysqlcluster.go [hunk 4]
```diff
@@ -270,6 +270,22 @@ func (c *MysqlCluster) EnsureVolumes() []corev1.Volume {
 			},
 		})
 	}
+	// add the ssl secret mounts
```
Comment (acekingke) [inline mysqlcluster/mysqlcluster.go:273]: comments ： Upcase the first letter, and add dot after statement compeleted.
-> `Add the ssl secret mounts.`

File: sidecar/init.go [hunk 5]
```diff
@@ -140,7 +140,10 @@ func runInitCommand(cfg *Config) error {
 	if err = copyFile(path.Join(mysqlCMPath, "my.cnf"), path.Join(mysqlConfigPath, "my.cnf")); err != nil {
 		return fmt.Errorf("failed to copy my.cnf: %s", err)
 	}
-
+	//ssl settins
```
Comment (acekingke) [inline sidecar/init.go:143]: add a blank line

File: sidecar/init.go [hunk 6]
```diff
@@ -140,7 +140,10 @@ func runInitCommand(cfg *Config) error {
...
+	if exists, _ := checkIfPathExists(utils.TlsMountPath); exists {
+		buildSSLdata(uid, gid)
+	}
```
Comment (acekingke) [inline sidecar/init.go:146]: Add a blank line

**Comment by runkecheng** [comment]
Before the first word of all comments, there must be a space. If the first word is not a variable name or function name, the first letter should be capitalized.

```
var test string

//test    ×
// test     √

// if test xxx    ×
// If test xxx    √
```

**Comment by runkecheng** [comment]
Before the first word of all comments, there must be a space. If the first word is not a variable name or function name, the first letter should be capitalized.

```
var test string

//test    ×
// test     √

// if test xxx    ×
// If test xxx    √
```

**Review by acekingke — APPROVED** [review]

**Review by runkecheng — APPROVED** [review]

**Pull request merged**